mypy/checkexpr.py

```diff
@@ -2445,9 +2445,14 @@ def has_member(self, typ: Type, member: str) -> bool:
         elif isinstance(typ, TypeType):
             # Type[Union[X, ...]] is always normalized to Union[Type[X], ...],
             # so we don't need to care about unions here.
-            if isinstance(typ.item, Instance) and typ.item.type.metaclass_type is not None:
-                return self.has_member(typ.item.type.metaclass_type, member)
-            if isinstance(typ.item, AnyType):
+            item = typ.item
+            if isinstance(item, TypeVarType):
```

Comment:
I wonder shouldn't we add the same fallback to upper bound at the very top of this function? For example if I have just
```python
def f(x: T) -> T:
    x + 0
```